static_frame/core/index_hierarchy.py

```diff
@@ -1513,42 +2204,46 @@ def fillna(self, value: tp.Any) -> 'IndexHierarchy':
             self._update_array_cache()
 
         blocks = self._blocks.fill_missing_by_unit(value, None, func=isna_array)
-        index_constructors = tuple(self._levels.index_types())
 
-        return self.__class__._from_type_blocks(blocks,
-                index_constructors=index_constructors,
+        return self.__class__._from_type_blocks(
+                blocks=blocks,
+                index_constructors=self._index_constructors,
                 name=self._name,
-                own_blocks=True
+                own_blocks=True,
                 )
 
-    def _sample_and_key(self,
+    def _sample_and_key(self: IH,
             count: int = 1,
             *,
             seed: tp.Optional[int] = None,
-            ) -> tp.Tuple['IndexHierarchy', np.ndarray]:
+            ) -> tp.Tuple[IH, np.ndarray]:
+        '''
+        Selects a deterministically random sample from this IndexHierarchy.
+
+        Returns:
+            The sampled IndexHierarchy
+            An integer array of sampled iloc values
+        '''
         if self._recache:
-            self._update_array_cache()
+            self._update_array_cache() # pragma: no cover
```

Comment:
Same here...

Reply:
See this response: https://github.com/InvestmentSystems/static-frame/pull/437#discussion_r831501537